### DISCUSSION TITLE
Is it possible to specify the side of the road?

### DISCUSSION THREAD

**Discussion by mertbakir** [valhalla/valhalla#3968] [Q&A]
Hello, I recently asked a question tomtom forum. Valhalla also generates a similar route, it goes straight south (down side on the image) on the same street, but POIs are located on both side of the street. I wonder if we can tell valhalla to route the related side of the street where the POI is located.

https://devforum.tomtom.com/t/routing-api-how-to-specificy-side-of-road/2891/5

**Comment by kevinkreiser** [comment]
we do have that capability but only for singly digitized roads. of you check the docs you will see we have `prefered_side` on the location object. let me know if that sounds like what you're looking for.

**Reply by mertbakir** [reply]
Thanks, I'll look into it but what do you mean by singly digitized road?

**Reply by kevinkreiser** [reply]
https://en.wikipedia.org/wiki/Dual_carriageway

**Reply by mertbakir** [reply]
Another question: Same parameter doesn't seem to exists for Matrix API?

https://valhalla.github.io/valhalla/api/matrix/api-reference/

https://valhalla.github.io/valhalla/api/turn-by-turn/api-reference/

**Reply by kevinkreiser** [reply]
digitized or drawn on the map with 2 separate geometries even though everyone thinks of it as a single road

**Reply by kevinkreiser** [reply]
the location spec is used for the matrix api as well

**Reply by mertbakir** [reply]
> the location spec is used for the matrix api as well

Great, thanks! I'll test it out.

**Reply by mertbakir** [reply]
> digitized or drawn on the map with 2 separate geometries even though everyone thinks of it as a single road

Makes sense, if it's divided they are already like 2 different roads.

**Reply by kevinkreiser** [reply]
it would be a nice feature for a boulevard to be treated as a single carriageway for the sake of telling which side you want to arrive on in some scenarios.
